### ISSUE TITLE
TSdoc / JSdoc

### ISSUE THREAD

**Issue by ivikash** [ananthakumaran/tide#165]
Hey, I wanted to know what is a good way to insert jsdoc or tsdoc in typescript mode.

**Comment by josteink** [comment]
I'm not sure I follow. Could you rephrase that question?

Do you mean you find manually typing `/** documentation here */` cumbersome? Are you looking for generators for existing functions?

Some clarifications would be useful.

**Comment by ivikash** [comment]
Yes. I am looking for a behaviour like `/**` and press tab to expand to 
```
/**
 *
 */
``` 
and auto indent prefixed with `*`, something like
```
/**
 * Lorem Ipsum
 * Another Lorem Ipsum
 */
```

This will remove the manual efforts. Also if you navigate to `https://github.com/syl20bnr/spacemacs/tree/master/layers/%2Blang/javascript#documentation-js-doc` you will see some other useful jsdoc shortcuts. If we can get that it will be very useful too.

**Comment by ananthakumaran** [comment]
if you are using js2-mode it's already supported by M-j. The support of this is not great in typescript-mode, there is an open issue https://github.com/ananthakumaran/typescript.el/issues/41. 
I am closing this because it can't be fixed in tide mode.

**Comment by ivikash** [comment]
ok. Thanks.

**Comment by ananthakumaran** [comment]
@ivikash tide now supports jsdoc insertion see https://github.com/ananthakumaran/tide/pull/207

**Comment by ivikash** [comment]
Thanks @ananthakumaran :-)
